pkg/metrics/metrics.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("registration_metrics")
+var Reg *prometheus.Registry
+
+const (
+	MetricLabelRouted   = "Routed"
+	MetricLabelRejected = "Rejected"
+	MetricLabelListed   = "Listed"
+)
+
+// histogram with labels
+var (
+	// RegServProxyRouteHistogramVec measures the time taken by proxy before forwarding the request
+	RegServProxyRouteHistogramVec *prometheus.HistogramVec
+	// RegServProxyResponseHistogramVec measures the response time for either response or error from proxy when there is no routing
+	RegServProxyResponseHistogramVec *prometheus.HistogramVec
```

Comment:
You could make the variable names shorter by dropping the `RegServ` prefix:

```suggestion
	// ProxyRouteHistogramVec measures the time taken by proxy before forwarding the request
	ProxyRouteHistogramVec *prometheus.HistogramVec
	// ProxyResponseHistogramVec measures the response time from proxy when there is no routing
	ProxyResponseHistogramVec *prometheus.HistogramVec
```